File: Benchmarking/InputResponse.py
# input_output_gpio_test.py
# Set GPIO 17 HIGH if GPIO 27 is HIGH

import RPi.GPIO as GPIO
import time

INPUT_PIN = 27   # BCM GPIO 27 (input)
OUTPUT_PIN = 17  # BCM GPIO 17 (output)

GPIO.setmode(GPIO.BCM)
GPIO.setup(INPUT_PIN, GPIO.IN)
GPIO.setup(OUTPUT_PIN, GPIO.OUT)

try:
    while True:
        if GPIO.input(INPUT_PIN) == GPIO.HIGH:
            GPIO.output(OUTPUT_PIN, GPIO.HIGH)
        else:
            GPIO.output(OUTPUT_PIN, GPIO.LOW)
except KeyboardInterrupt:
    pass
finally:
    GPIO.cleanup()

File: Benchmarking/InputResponse.rs
// input_output_gpio_test.rs
// Mirror input GPIO 27 to output GPIO 17

use rppal::gpio::{Gpio, InputPin, OutputPin};
use std::error::Error;
use std::thread;
use std::time::Duration;

const INPUT_GPIO: u8 = 27;  // Input pin
const OUTPUT_GPIO: u8 = 17; // Output pin

fn main() -> Result<(), Box<dyn Error>> {
    let gpio = Gpio::new()?;

    let input: InputPin = gpio.get(INPUT_GPIO)?.into_input();
    let mut output: OutputPin = gpio.get(OUTPUT_GPIO)?.into_output();

    loop {
        if input.is_high() {
            output.set_high();
        } else {
            output.set_low();
        }

        // Optionally reduce CPU usage
        //thread::sleep(Duration::from_micros(10));
    }
}
